File: Cleared_Dataset/HumanEval_Java/6_P13.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class ParseNestedParens {
    /**
     * Input to this function is a string represented multiple groups for nested parentheses separated by spaces.
     * For each of the group, output the deepest level of nesting of parentheses.
     * E.g. (()()) has maximum two levels of nesting while ((())) has three.

     * >>> parse_nested_parens('(()()) ((())) () ((())()())')
     * [2, 3, 1, 3]
     * @param parenString a string represented by multiple groups for nested parentheses separated by spaces.
     * @return a list of integers where each integer denotes the deepest level of nesting of parentheses of each group in the input
     *
     *
     */
    public static List<Integer> parseNestedParens(String parenString) {

File: Cleared_Dataset/HumanEval_Py/6_P13.py
from typing import List


def parse_nested_parens(paren_string: str) -> List[int]:
    """ Input to this function is a string represented multiple groups for nested parentheses separated by spaces.
    For each of the group, output the deepest level of nesting of parentheses.
    E.g. (()()) has maximum two levels of nesting while ((())) has three.

    args:
        paren_string: string containing multiple groups of nested parentheses separated by spaces
    
    returns:
        list of integers representing the deepest level of nesting of parentheses for each group
    
    examples:
    >>> parse_nested_parens('(()()) ((())) () ((())()())')
    [2, 3, 1, 3]
    """
